velox/substrait/SubstraitToVeloxPlan.cpp

```diff
@@ -84,176 +84,76 @@ core::PlanNodePtr SubstraitVeloxPlanConverter::toVeloxPlan(
-        toVeloxType(aggOutType->type), std::move(aggParams), funcName);
+        aggVeloxType, std::move(aggParams), funcName);
     aggExprs.emplace_back(aggExpr);
-
-    // Initialize the Aggregate Step.
-    if (!phaseInited) {
-      auto phase = aggFunction.phase();
-      switch (phase) {
-        case ::substrait::AGGREGATION_PHASE_INITIAL_TO_INTERMEDIATE:
-          aggStep = core::AggregationNode::Step::kPartial;
-          break;
-        case ::substrait::AGGREGATION_PHASE_INTERMEDIATE_TO_INTERMEDIATE:
-          aggStep = core::AggregationNode::Step::kIntermediate;
-          break;
-        case ::substrait::AGGREGATION_PHASE_INTERMEDIATE_TO_RESULT:
-          aggStep = core::AggregationNode::Step::kFinal;
-          break;
-        case ::substrait::AGGREGATION_PHASE_INITIAL_TO_RESULT:
-          aggStep = core::AggregationNode::Step::kSingle;
-          break;
-        default:
-          VELOX_NYI("Substrait conversion not supported for phase '{}'", phase);
-      }
-      phaseInited = true;
-    }
-    outIdx += 1;
   }
 
-  // Construct the Aggregate Node.
   bool ignoreNullKeys = false;
-  std::vector<core::FieldAccessTypedExprPtr> preGroupingExprs;
-  if (projectOutNames.size() == 0) {
-    // Conduct Aggregation directly.
-    std::vector<std::string> aggOutNames;
-    aggOutNames.reserve(aggMeasureSize);
-    for (int idx = 0; idx < aggMeasureSize; idx++) {
-      aggOutNames.emplace_back(
-          substraitParser_->makeNodeName(planNodeId_, idx));
-    }
-    return std::make_shared<core::AggregationNode>(
-        nextPlanNodeId(),
-        aggStep,
-        veloxGroupingExprs,
-        preGroupingExprs,
-        aggOutNames,
-        aggExprs,
-        aggregateMasks,
-        ignoreNullKeys,
-        childNode);
-  } else {
-    // A Project Node is needed before Aggregation.
-    auto projectNode = std::make_shared<core::ProjectNode>(
-        nextPlanNodeId(),
-        std::move(projectOutNames),
-        std::move(projectExprs),
-        childNode);
-    std::vector<std::string> aggOutNames;
-    aggOutNames.reserve(aggMeasureSize);
-    for (int idx = 0; idx < aggMeasureSize; idx++) {
-      aggOutNames.emplace_back(
-          substraitParser_->makeNodeName(planNodeId_, idx));
-    }
-    return std::make_shared<core::AggregationNode>(
-        nextPlanNodeId(),
-        aggStep,
-        veloxGroupingExprs,
-        preGroupingExprs,
-        aggOutNames,
-        aggExprs,
-        aggregateMasks,
-        ignoreNullKeys,
-        projectNode);
+  std::vector<std::shared_ptr<const core::FieldAccessTypedExpr>> aggregateMasks(
+      sAgg.measures().size());
+  std::vector<std::shared_ptr<const core::FieldAccessTypedExpr>>
+      preGroupingExprs = {};
+
+  // Get the output names of Aggregation.
+  std::vector<std::string> aggOutNames;
+  aggOutNames.reserve(sAgg.measures().size());
+  for (int idx = groupingOutIdx; idx < groupingOutIdx + sAgg.measures().size();
```

Comment:
Use veloxGroupingExprs.size() for groupingOutIdx

Reply:
OK, it is fixed.